index.ts

```diff
@@ -271,7 +271,7 @@ function ensureTypeScriptInstance(options: Options, loader: any): { instance?: T
 
     // Create the TypeScript language service
     var servicesHost = {
-        getScriptFileNames: () => Object.keys(files),
+        getScriptFileNames: () => Object.keys(files).filter(filePath => !!filePath.match(/(\.d)?\.tsx?$/)),
```

Comment:
Oh, also, I'd use `Regexp.prototype.test` instead of `match`, it's always boolean - `/\.tsx?$/i.test(filePath)`. I added the `i` flag.